app/assets/javascripts/homepage/conversation.coffee

```diff
@@ -0,0 +1,97 @@
+conversation = angular.module("Conversation", ["ngRoute", "SharedServices"])
+
+conversation.config(["$routeProvider", ($routeProvider) ->
+    $routeProvider.when("/conversation/:id", {
+        templateUrl: "/assets/conversation.html",
+        controller: "ConversationController"
+    })
+
+]).directive("onEscape", ->
+    (scope, element, attributes) ->
+        element.bind("keydown keypress", (event) ->
+            if event.which == 27
+                scope.$apply(-> scope.$eval(attributes.onEscape))
+                event.preventDefault()
+        )
+).controller("ConversationController", ["$scope", "$routeParams", "SharedRequests", "TimeUtil", ($scope, $routeParams, SharedRequests, TimeUtil) ->
+    # Constants
+    POST_SUBMISSION_TIMEOUT_PERIOD_MS = 5000
+    # State variables
+    conversationId = parseInt($routeParams["id"])
+    currentUserId = 1 # TODO Fetch current user through login service.
+    currentlyEditingPost = false
+    postSubmissionInProgress = false
+    conversation = {}
+    postsById = {}
+    currentPostEditId = null
+    postSubmissionTimer = null
+    # Scope methods and models
+    $scope.editPostText = ""
+    $scope.postSubmitError = ""
+    $scope.timeElapsedMessage = (timestamp) -> TimeUtil.timeIntervalAsString(TimeUtil.timeSince1970InSeconds() - TimeUtil.timeFromTimestampInSeconds(timestamp)) + " ago"
+    $scope.postEditorClass = -> if currentlyEditingPost then "" else "hidden"
+    $scope.postLengthClass = -> if currentlyEditingPost then "posts-shortened" else "posts-full-length"
+    $scope.addPostClicked = ->
+        $scope.postSubmitError = ""
+        currentlyEditingPost = true
+        $scope.editPostText = "" if currentPostEditId # Clear the editor if the user was previously editing an existing post.
+        currentPostEditId = null
+        dispatchFocusTextarea()
+
+    $scope.cancelPostClicked = -> currentlyEditingPost = false
```

Comment:
A comment for all the somethingClicked methods: might as well call them somethingButtonClicked to make it more clear.